### DISCUSSION TITLE
Differing values in component

### DISCUSSION THREAD

**Discussion by hammo92** [pmndrs/jotai#955] [Q&A] [answered]
I'm new to Jotai, apologies if this is a basic question.

When using an atom in a component I'm getting differing values inside and outside of the onChange function. The value outside the function updates with the slider, but inside the function doesn't. This is using Next JS if that makes any difference.

![carbon(1)](https://user-images.githubusercontent.com/2330923/149154805-fd8a574c-3a9f-4ec3-9132-ade8f48b3e12.png)
:
![functionCap](https://user-images.githubusercontent.com/2330923/149155366-ad45a9e1-50a4-47c7-aeed-591861f3397d.PNG)

**Comment by Thisen** [comment, marked as answer]
`useAtom` works similarly to `useState`. It's asynchronous and therefore state won't be changed until the next render. This means you can't expect to get the state value, right after you've set it.

**Reply by hammo92** [reply]
wouldn't this still mean that both the console.log inside the onChange function and outside of the onChange function would return the same result? What is confusing is that the value of the atom inside the onChange function never updates.

**Reply by Thisen** [reply]
>wouldn't this still mean that both the console.log inside the onChange function and outside of the onChange function would return the same result?

No, not exactly. `setStrokeWidth` is asynchronous and the new value will be set on the next render.

>What is confusing is that the value of the atom inside the onChange function never updates.

I can't speak to the exact case, since I don't know the `<Slider />` component, but I'm assuming it's called multiple times before the render has occurred. Therefore you won't see a different value. As mentioned, the value is updated on render.

**Reply by hammo92** [reply]
The problem seems to be with the component, as switching it out for a standard range input doesn't have the same issue. Should probably have tested that first! Thanks for the help

**Reply by Thisen** [reply]
Happy to help 🎉
